docs/user-manual/core-tools.md

```diff
@@ -219,12 +219,354 @@ As c++ is a strongly typed language, the signal template parameters signature ha
+  
+  <!-- Constant in a text -->
+  <quote> This is movie reference : ${Answer} </quote>
+  
+</dqm4hep>
+```
+
+### MySQL database parameter select
+
+To be done - requires documentation on MySQL parameter database and tables ...
+
+### Use of environment variables
+
+Constants are a nice feature of the DQM4hep XML parser but can sometimes be a problem when users are dealing with passwords and other private sensitive data such API tokens. Environment variables are often use to deal with this problem as they are defined locally in your shell and thus not accessible you opening and reading the XML file in an editor. The XML parser can use environment variables by using a similar syntax as for constants (see above) with `$ENV{var}` where `var` is an environment variable.
+
+Example:
+
+```xml
+<dqm4hep>
+  <!-- Oops ! Password hardcoded... -->
+  <login user="Arthur" password="Cuillere"/>
+</dqm4hep>
+```
+
+then use an environment variable instead:
+
+```xml
+<dqm4hep>
+  <login user="Arthur" password="$ENV{LOGIN_PASS}"/>
+</dqm4hep>
+```
+
+<div class="warning-msg">
+  <i class="fa fa-warning"></i>
+  Don't forget to export the referenced environment variables before using the XML parser as you will get an exception if they are not defined !
+</div>
+
+### XML for loops
+
+The element `<for>` allows to run a simple for loop and duplicate XML elements. To keep it simple, the loop is run over integer id (`id`) from a lower value (`begin`) to an upper value (`end`) with a user defined increment (`increment`). A for loop is run over the whole content of the `<for>` element, look for the string "$FOR{*id*}" and replace it by the current loop value. The replacement operates on:
+
+- the XML comments content
+- all the XML element attributes
+- the XML texts
+
+Example:
+```xml
+<dqm4hep>
+  <for id="count" begin="0" end="3" increment="1">
+    <!-- The current value of the counter is $FOR{count} -->
+    <counter value="$FOR{count}"/>
+  </for>
+</dqm4hep>
+```
+
+will be replaced by:
+```xml
+<dqm4hep>
+  <!-- The current value of the counter is 0 -->
+  <counter value="0"/>
+  <!-- The current value of the counter is 1 -->
+  <counter value="1"/>
+  <!-- The current value of the counter is 2 -->
+  <counter value="2"/>
+  <!-- The current value of the counter is 3 -->
+  <counter value="3"/>
+</dqm4hep>
+```
+
+By construction, it is also possible the run nested for loops by changing the loop id:
+
+```xml
+<dqm4hep>
+  <for id="x" begin="0" end="2" increment="1">
+    <for id="y" begin="0" end="2" increment="1">
+      <entry x="$FOR{x}" y="$FOR{y}"/>
+    </for>
+  </for>
+</dqm4hep>
+```
+
+will be replaced by:
+
+```xml
+<dqm4hep>
+  <entry x="0" y="0"/>
+  <entry x="0" y="1"/>
+  <entry x="0" y="2"/>
+  <entry x="1" y="0"/>
+  <entry x="1" y="1"/>
+  <entry x="1" y="2"/>
+  <entry x="2" y="0"/>
+  <entry x="2" y="1"/>
+  <entry x="2" y="2"/>
+</dqm4hep>
+```
+
+The `<for>` element attributes possible values are summurized in table below. 
```

Comment:
** Possible values for the `<for>` element attribute are summarized in the table below. **

Reply:
OK